EngineLayer/SpectralLibrarySearch/SpectralLibrary.cs

```diff
@@ -330,6 +351,120 @@ private LibrarySpectrum ReadLibrarySpectrum(StreamReader reader, bool onlyReadHe
                     //TODO: figure out terminus
                     FragmentationTerminus terminus = (FragmentationTerminus)Enum.Parse(typeof(FragmentationTerminus), "None", true);
 
+                    //TODO: figure out amino acid position
+                    var product = new Product(peakProductType, terminus, experMz, fragmentNumber, 0, neutralLoss);
+
+                    matchedFragmentIons.Add(new MatchedFragmentIon(ref product, experMz, experIntensity, fragmentCharge));
+                }
+            }
+
+            return new LibrarySpectrum(sequence, precursorMz, z, matchedFragmentIons, rt);
+        }
+
+        private LibrarySpectrum ReadLibrarySpectrum_pDeep(StreamReader reader, bool onlyReadHeader = false)
+        {
+            char[] nameSplit = new char[] { '/', '_' };
+            char[] mwSplit = new char[] { ':' };
+            char[] commentSplit = new char[] { ' ', ':', '=' };
+            char[] modSplit = new char[] { '/', '(', ')' };
+            char[] fragmentSplit = new char[] { '\t', '/' };
```

Comment:
these arrays should be static variables for the class. it will save you some time garbage collecting old arrays. same w/ arrays in `ReadLibrarySpectrum`